### DISCUSSION TITLE
Clarification: How to convert frontmatter tags

### DISCUSSION THREAD

**Discussion by tobiasvl** [ObsidianPublisher/obsidian-github-publisher#48] [Q&A — Plugin] [answered]
Hello! This is probably a stupid question, but I'm unable to figure out the answer, so in some way I guess I consider it a bug in the documentation.

`README.md` says the following:

> Convert frontmatter/inline field into tags : This will convert the value associated to preconfigured field into frontmatter tags. You can also prevent some value to be converted with the second field. Note: If the value is a link, the converted value will be the filename or the displayed name. You can either exclude the filename or the displayed name.

(However, the relevant setting is not called "Convert frontmatter/inline field into tags", it's called "Convert frontmatter/dataview field to tags". That should probably be fixed.)

The way I interpret this option, my file in Obsidian can have the following frontmatter:

```
---
jekyll-tags:
  - Foo Bar
  - Baz Quux
---
```

And then, if I set "Convert frontmatter/dataview field to tags" to "jekyll-tags", the resulting file on GitHub should get the following frontmatter:

```
---
tags:
  - Foo Bar
  - Baz Quux
---
```

But that doesn't happen, so I'm clearly misunderstanding what this option does. Can I make it do what I want, or if not, what does this option actually do?

(Background: The reason I want to do this is that I'm trying to publish from Obsidian to GitHub Pages with this plugin, and Jekyll allows spaces in tags while Obsidian does not, so I want to separate Obsidian tags from Jekyll tags in Obsidian but use only Jekyll tags for GitHub Pages.)

Thanks!

**Comment by tobiasvl** [comment]
I see now that the plugin has a "Text replacer" option, so what I want to do is probably just to use that to replace "jekyll-tags" with "tags". That seems the simpler option. Still, it doesn't really clear up for me what the "Convert frontmatter/dataview field to tags" option does!

**Comment by Mara-Li** [comment, marked as answer]
Hello!
This option allow you to convert a dataview entry to a tag.

For example, you have in a dataview inline with this : 
```md
---
tags: [Roleplay, TTRPG]
title: Another idea for NPC
---
type:: NPC
```
The `PNJ` value will be added in the tags, as that : 
```md
---
tags: [Roleplay, TTRPG, NPC]
---
type:: NPC
```

Same for a frontmatter entry like that : 
```md
---
type: NPC
---
```
That will turn into : 
```md
---
tags: [NPC]
type: NPC
---
```

This function can be really usefull in this example : 
```md
---
tags: [Tutorial]
---
linked:: [[wordmap]]
```

→  
```md
---
tags: [Tutorial, worldmap]
---
linked:: [[worldmap]]
```

**Comment by tobiasvl** [comment]
Aha! Thank you. I've never used Dataview, so it was a bit confusing to me that the option was called "Convert frontmatter/dataview field to tags"; the slash makes it seem like it's for either frontmatter or dataview, but it sounds like it's actually for either dataview frontmatter or inline dataview?

**Reply by Mara-Li** [reply]
Yes, you can use it for both of them!
